### PR TITLE
RSX improvements and gl fix

### PR DESCRIPTION
**Pull request by jarveson**
This fixes arguments / method_registers in nv3089_image_in, 
also convert_swizzle.elf now runs fully for all textures/sizes on d12, fixing #1113 

The new convert_linear_swizzle was tested to output the same as the reference loop with various parameters https://github.com/jarveson/rpcs3elfs/tree/master/SwizzleSpeedTest 

Let me know if theres any issues/questions!

### CODE REVIEW

File: rpcs3/Emu/RSX/RSXThread.h [hunk 1]
```diff
@@ -121,10 +121,86 @@ namespace rsx
...
+
 	u32 get_address(u32 offset, u32 location);
-	u32 linear_to_swizzle(u32 x, u32 y, u32 z, u32 log2_width, u32 log2_height, u32 log2_depth);
 
-	u32 get_vertex_type_size(u32 type);
+	template<typename T>
+	void pad_texture(void* inputPixels, void* outputPixels, u16 inputWidth, u16 inputHeight, u16 outputWidth, u16 outputHeight) {
+		T *src, *dst;
+		src = (T *)(inputPixels);
+		dst = (T *)(outputPixels);
```
Comment (danilaml) [inline rpcs3/Emu/RSX/RSXThread.h]: any reason these are C-style casts?

**Comment by vlj** [comment]
Nice work! The unswizzling algorithm is more cache friendly this way. 

Some stylistic remark:
Brace should go on the next line.

**Comment by vlj** [comment]
Temporary variable shouldn't use camel style ie write temp_swizzled instead of tempSwizzled

File: rpcs3/Emu/RSX/RSXThread.h [hunk 2]
```diff
@@ -121,10 +121,86 @@ namespace rsx
 
 	extern u32 method_registers[0x10000 >> 2];
 
+    u32 get_vertex_type_size(u32 type);
```
Comment (vlj) [inline rpcs3/Emu/RSX/RSXThread.h]: Missing tab

**Comment by PZerua** [comment]
This PR allows Machinarium to go ingame, with broken textures:

machinarium

File: rpcs3/Emu/RSX/RSXThread.h (hunk 1 above)
Comment (jarveson) [inline rpcs3/Emu/RSX/RSXThread.h]: Nope, no reason, i've changed them to static_cast's now

**Comment by jarveson** [comment]
I think i got all the style changes taken care of now

**Comment by tambry** [comment]
@jarveson Needs to be rebased.

**Comment by jarveson** [comment]
rebased!

**Pull request merged**